src/lib/raft/raft.c

```diff
@@ -158,6 +159,32 @@ raft_add_vote(struct raft *raft, int src, int dst)
 	return true;
 }
 
+/**
+ * Vote can be revoked if it was volatile (not yet in WAL) and thus wasn't shown
+ * to any other instance yet.
+ */
+static void
+raft_del_vote(struct raft *raft, int src, int dst)
+{
+	struct raft_vote *v = &raft->votes[src];
+	assert(v->did_vote);
+	v->did_vote = false;
+	assert(raft->voted_count > 0);
+	--raft->voted_count;
+	v = &raft->votes[dst];
+	assert(v->count > 0);
+	bool was_max = v->count == raft->max_vote;
+	--v->count;
+	if (was_max) {
+		raft->max_vote = 0;
```

Comment:
A tiny optimisation: raft->max_vote is >= previous_max - 1 anyway.

Reply:
`raft->max_vote--` instead of `raft->max_vote = 0`.
If you were max and did `--v->count` it means either max_vote is still the same or it is decremented.

Reply:
But we still need to do the fullscan, because we don't know if it changed or not. Anyway, updated to your version.

Reply:
Yeah, we do, but we'll perform less assignments.

There's another one: you may break the loop as soon as max_vote reaches old_max (basically as soon as max_vote is increased).
But this is over the top, I guess. Will make the code harder to follow. Current version looks fine.